app/grandchallenge/archives/models.py

```diff
@@ -267,6 +268,14 @@ def interface_viewname(self):
     def civ_sets_related_manager(self):
         return self.items
 
+    @property
+    def interfaces(self):
+        return ComponentInterface.objects.filter(
+            slug__in=self.items.exclude(values__isnull=True).values_list(
+                "values__interface__slug", flat=True
+            )
+        ).distinct()
+
```

Comment:
This query could also be: `ComponentInterface.objects.filter(componentinterfacevalue__archive_items__in=self.items).exclude(componentinterfacevalue__isnull=True)`

I'm not sure if that is more efficient, or if the difference is significant. 

In your proposed query, I would switch to filtering by `pk` because that is more efficient than filtering by `slug`. 

And you can move this property to the `ValuesForInterfacesMixin`, saves a few lines of code ;-)

Reply:
In our discussion about this last week, we agreed not to go through `componentinterfacevalue__` since that table is huge and thus the query not efficient. I prefer to stick with this one, and agree with using `pk` instead of the `slug`.
Also, moved the property to `ValuesForInterfacesMixin`

Done.